osu.Game.Tests/Visual/Multiplayer/TestSceneLoungeRoomsContainer.cs

```diff
@@ -107,21 +109,21 @@ public void TestStringFiltering()
         [Test]
         public void TestRulesetFiltering()
         {
-            AddRooms(2, new OsuRuleset().RulesetInfo);
-            AddRooms(3, new CatchRuleset().RulesetInfo);
+            AddStep("add rooms", () => RoomManager.AddRooms(2, new OsuRuleset().RulesetInfo));
+            AddStep("add rooms", () => RoomManager.AddRooms(3, new CatchRuleset().RulesetInfo));
 
+            // Todo: What even is this case...?
+            AddStep("set empty filter criteria", () => container.Filter(null));
```

Comment:
This extra step being seemingly needed is actually a bug manifesting in `RoomsContainer`. See:

https://github.com/ppy/osu/blob/f2482e354e9557c9e8be3449c9c958c779f6f216/osu.Game/Screens/OnlinePlay/Lounge/Components/RoomsContainer.cs#L96

`OnlinePlayTestSceneDependencies` by default supply `new FilterCriteria()` and not `null`, leading to the failure because the check above does not check for `criteriaRuleset == null`. It should probably do so.

Reply:
What I meant by this comment is that `null` is a weird case to have in the first place. I'll add a `criteria.Ruleset == null` check since you're right (it doesn't change this test) but the `FilterCriteria` itself should never be null.

I wrote a todo because this may/will probably be removed as a valid test case in the future.